broker/api.go

```diff
@@ -10,9 +10,16 @@ type PermittedAccount struct {
 	PermittedRoleName []string
 }
 
+type AWSCredentialsJSON struct {
+	SessionId    string `json:"sessionId"`
+	SessionKey   string `json:"sessionKey"`
+	SessionToken string `json:"sessionToken"`
+}
+
 type Broker interface {
 	UpdateConfiguration(config *configuration.Configuration) error
 	GetUserAllowedAccounts(username string) ([]PermittedAccount, error)
 	IsUserAllowedToAssumeRole(username string, accountName string, roleName string) (bool, error)
 	GetConsoleURLForAccountRole(accountName string, roleName string, username string) (string, error)
+	GenerateTokenCredentials(accountName string, roleName string, username string) (*AWSCredentialsJSON, error)
```

Comment:
Please return a `TokenWriter` instead.

Reply:
On reading future core, this might not be as clean. On this pull request it seems you can just write the output, however when generating human consumable output (html) we would need to potentially pass also the templates and would require is to put template code here. I thing the issue is the name: AWSCredentialsJSON should be instead STSCredentials. What do you think?

Reply:
It seems to me that "STS" is still an AWS-specific term.
However, the more important point in designing this API is that the return value should be an interface type and not a struct type, so that this code has no dependencies on specific implementations. If the `TokenWriter` interface name seems too limited, we can pick a better name.